xarray/core/dataset.py

```diff
@@ -5877,15 +6101,42 @@ def ffill(self: T_Dataset, dim: Hashable, limit: int | None = None) -> T_Dataset
         Parameters
         ----------
         dim : Hashable
-            Specifies the dimension along which to propagate values when
-            filling.
+            Specifies the dimension along which to propagate values when filling.
         limit : int or None, optional
             The maximum number of consecutive NaN values to forward fill. In
             other words, if there is a gap with more than this number of
             consecutive NaNs, it will only be partially filled. Must be greater
             than 0 or None for no limit. Must be None or greater than or equal
             to axis length if filling along chunked axes (dimensions).
 
+        Examples
```

Comment:
Again there are keyword arguments that will affect the result. Try to come up with a dataset which will give different results for different values of the keyword arguments. For example if in one place there was a lone NaN, but somewhere else there was a string of 3 consecutive NaNs, then limit=1 vs limit=3 would matter for the result.

That way you don't have to make a new dataset for each option you want to show.